library/src/main/java/com/novoda/storagepathfinder/ExternalFileDirectoryInspector.java

```diff
@@ -93,6 +93,15 @@ private boolean isValidPathForSecondaryStorage(String absolutePath, Filter base)
         if (base == Filter.APPLICATION) {
             pathToCompare = getDirectoryPathAboveTheAndroidFolderFrom(new File(absolutePath));
         }
-        return !pathToCompare.isEmpty() && !pathToCompare.equals(primaryStoragePath);
+        return !pathToCompare.isEmpty() && !isSameAsPrimaryStoragePath(pathToCompare);
+    }
+
+    private boolean isSameAsPrimaryStoragePath(String pathToCompare) {
+        for (StoragePath primaryStoragePath : primaryStoragePaths) {
+            if (pathToCompare.equals(primaryStoragePath.getPathAsString())) {
```

Comment:
Does this resolve symlinks?

Reply:
You might want to use `File.getCanonicalPath()` to be sure

Reply:
9cefcc4